processor/spanmetricsprocessor/processor.go

```diff
@@ -282,10 +300,26 @@ func (p *processorImp) collectCallMetrics(ilm pdata.InstrumentationLibraryMetric
 		dpCalls.SetTimestamp(pdata.NewTimestampFromTime(time.Now()))
 		dpCalls.SetIntVal(p.callSum[key])
 
-		p.metricKeyToDimensions[key].CopyTo(dpCalls.Attributes())
+		if dimensions, ok := p.getDimensionsByMetricKey(key); ok {
+			dimensions.CopyTo(dpCalls.Attributes())
+		}
 	}
 }
 
+// getDimensionsByMetricKey gets dimensions from `metricKeyToDimensions` cache.
+func (p *processorImp) getDimensionsByMetricKey(k metricKey) (*pdata.AttributeMap, bool) {
```

Comment:
It looks like this should actually be:
```suggestion 
func (p *processorImp) getDimensionsByMetricKey(k metricKey) (*pdata.AttributeMap, error)
```
Each time the result is false, an error is logged, so why not return instead?